dvc/parsing/interpolate.py

```diff
@@ -80,6 +81,31 @@ def _(obj: bool):
     return "true" if obj else "false"
 
 
+@to_str.register(dict)
+def _(obj: dict):
+    result = ""
+    for k, v in flatten(obj).items():
+        if isinstance(v, bool):
```

Comment:
This only works for `argparse.BooleanOptionalAction` not for `store_true/false`. Maybe need some better way to handle this?

Reply:
The same issue about different options, as commented below https://github.com/iterative/dvc/pull/7907#discussion_r903429914 

Which option do you think we should consider as the most appropriate default? 

Note that this can be used for things beyond `argparse`, for example you can use the interpolation to pass flags to arbitrary executables that you call inside `cmd`

Reply:
I also have no idea about it.

Reply:
I think that's more than enough to support for now.